lib/proxy_api/bmc.rb

```diff
@@ -5,6 +5,7 @@ class BMC < ProxyAPI::Resource
     def initialize(args)
       @target = args[:host_ip] || '127.0.0.1'
       @url = args[:url] + "/bmc"
+      @provider = args[:bmc_provider]
```

Comment:
Why exactly we need to pass BMC provider into API queries? It's something that is configured on smart-proxy via `YAML` configuration, or at least that's I'd expect.

Reply:
Not all hardware supports Redfish, so you might use IPMI for one server and Redfish for the other.

Reply:
That's exactly it. We have some small number of hosts that are too old to support Redfish. They're going away, but they're not gone yet.

Reply:
There was also this earlier change, which only got us part of the way there -- 
https://projects.theforeman.org/issues/27207

Reply:
Look who filed the issue and fixed it :-) Why I am not surprised...

Makes sense.